### PR TITLE
Extend maintenance to support minor updates

### PR DESCRIPTION
**Pull request by Kidswiss**
## Summary

* Creation of maintenance job moved from P+T composition to comp functions
* The job now dynamically determines if a minor update is available and will trigger it
* If no minor update is available a normal security maintenance will ocurr

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.
- [x] Link this PR to related issues.

### CODE REVIEW

File: pkg/maintenance/postgresql.go [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+	}
+
+	p.log.Info("Found versions", "current", currentVersion, "latest", latestMinor)
+	if currentVersion != latestMinor {
+		p.log.Info("Doing a minor upgrade")
+		return p.createMinorUpgrade(clusterName, latestMinor)
+	}
+
+	p.log.Info("Checking for EOL")
+	if versionList != nil && p.isEOL(currentVersion, *versionList) {
```
Comment (zugao) [inline pkg/maintenance/postgresql.go:97]: Is this related with the operator update? I mean the version list will remain always the same unless we upgrade the operator?

Comment (Kidswiss) [reply pkg/maintenance/postgresql.go:97]: Yes, if we update the operator then it will set the EOL accordingly. We currently don't send mails for that.

File: pkg/maintenance/postgresql.go [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+			SgCluster:  clusterName,
+			Op:         opName,
+			MaxRetries: pointer.Int(1),
+		},
+	}
+
+}
+
+func (p *PostgreSQL) applyDbOps(obj *stackgresv1.SGDbOps) error {
+	err := p.Client.Delete(p.ctx, obj)
```
[the remark this answers is not in the document]
Comment (Kidswiss) [reply pkg/maintenance/postgresql.go:230]: Do we want to retain a history here? I would rather not keep a pile of old objects and jobs around to be honest.

**Comment by Kidswiss** [comment]
> Why we don't create the service account, role and role binding directly in P+T? We should always have some maintenance on even if the customer does not specify one so it makes sense to have it .

It used to be a P+T composition, but I changed it to a comp function. I find it much more comfortables. Also the whole cron parsing in the P+T composition was just horrible. By moving this to the comp-functions the P+T composition is reduced by about 700+ lines.

File: pkg/maintenance/postgresql.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (zugao) [reply pkg/maintenance/postgresql.go:230]: I also am against having a pile of SGDbOps manifests around but we should consider if we need `some` history.

**Comment by zugao** [comment]
> It used to be a P+T composition, but I changed it to a comp function. I find it much more comfortables. Also the whole cron parsing in the P+T composition was just horrible. By moving this to the comp-functions the P+T composition is reduced by about 700+ lines.

Going into that direction you could argue to move everything in comp-functions :D

**Review by zugao — APPROVED** [review]
LGTM

**Comment by Kidswiss** [comment]
> > It used to be a P+T composition, but I changed it to a comp function. I find it much more comfortables. Also the whole cron parsing in the P+T composition was just horrible. By moving this to the comp-functions the P+T composition is reduced by about 700+ lines.
> 
> Going into that direction you could argue to move everything in comp-functions :D

Yes I would argue for that actually :) But we can move one thing at a time, as I just did.

File: pkg/maintenance/postgresql.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kidswiss) [reply pkg/maintenance/postgresql.go:230]: I'd say we wait if the need arises for having a history. We could probably save it in an array of the claim's status and then show the last 5 maintenances or something.
